pycqed/measurement/waveform_control/pulsar.py

```diff
@@ -1275,6 +1613,15 @@ def stop(self):
             self._stop_awg(awg)
     
     def program_awgs(self, sequence, awgs='all'):
+        try:
+            self._program_awgs(sequence, awgs)
+        except Exception as e:
+            log.warning(f'Pulsar: Exception {repr(e)} while programming AWGs. '
+                        f'Retrying after resetting the sequence cache.')
+            self.reset_sequence_cache()
+            self._program_awgs(sequence, awgs)
```

Comment:
The try-except doesn't really hurt here, but is this still needed?

Reply:
Since sequence caching is am optional speedup feature, it makes sense to fall back to the safer version of not using the cache if something goes wrong. This retry is relatively cheap and might prevent crashed measurements, so I would keep it.